assoc-aggregate/assoc-aggregate.wdl

```diff
@@ -491,19 +491,18 @@ task sbg_prepare_segments_1 {
 		import os
 		import shutil
 		import datetime
+		import logging
+
+		if "~{debug}" == "true":
+			logging.basicConfig(level=logging.DEBUG)
+		else:
+			logging.basicConfig(level=logging.INFO)
 
 		def find_chromosome(file):
+			# Corresponds with find_chromosome() in CWL
 			chr_array = []
 			chrom_num = split_on_chromosome(file)
-			if len(chrom_num) == 1:
-				acceptable_chrs = [str(integer) for integer in list(range(1,22))]
-				acceptable_chrs.extend(["X","Y","M"])
-				if chrom_num in acceptable_chrs:
-					return chrom_num
-				else:
-					print("%s appears to be an invalid chromosome number." % chrom_num)
-					exit(1)
-			elif (unicode(str(chrom_num[1])).isnumeric()):
+			if (unicode(str(chrom_num[1])).isnumeric()):
 				# two digit number
 				chr_array.append(chrom_num[0])
```

Comment:
I wouldn't mess with this now, but there is no need to create an array `chr_array`. You could just return `chrom_num[0:1]`, or something to that extent.

Reply:
You are correct, but afaik that's what the CWL does. Something similar is also done in aggregate_list -- see comment on line 348 of the WDL where chromosomes_basename seems to get rewritten after we go to a lot of trouble to construct it.